testsuite/pytests/sli2py_mpi/test_issue_1974.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+#
+# test_issue_1974.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from mpi_test_wrapper import MPITestAssertEqual
+
+
+@MPITestAssertEqual([1, 2, 4], debug=True)
+def test_issue_1974():
+    """
+    For each of the two parrot neurons, exactly one connection from/to the in/out MUSIC proxy must be created,
+    specifically to the proxy on the rank on which the parrot neuron exists. Therefore, the pooled connections across
+    ranks must be identical independent of number of ranks.
+    """
+
+    import nest
+
+    if not nest.ll_api.sli_func("statusdict/have_music ::"):
+        return
+
+    nest.total_num_virtual_procs = 4
```

Comment:
@JanVogelsang  This line is not in the SLI version. Can you check if the test passes when you remove it? Then we could port this test. At the same time, could you then create a follow-up issue about MUSIC and threading not working fully?

Reply:
Oh you are right, strange, not sure why I added this line.

The script (i.e., runner.py) indeed doesn't fail without that line, which also makes sense of course, but the test itself still fails. How can I debug this?

Reply:
The issue is not related to MUSIC as it seems, I created a minimal reproducer (#3442 ).

Reply:
I don't have music installed on my computer. Could you run `runner.py` for 1, 2, 4 MPI processes and post the resulting OTHER files?

Reply:
Sure, I can't find the files though. Should they be written in the same temp directory where I find the runner.py file?

Reply:
The OTHER files look sane (identical if one would combine the files from all processes and sort them), but somehow the test still fails, even if not setting the number of VPs to 4.
If one explicitly sets the number of VPs to 4, the original bug is now fixed with PR #3445, but now there is another issue, which I described in issue #3446.